src/JiraStatusNotifier/Channel/TicketsByAssignee.php

```diff
@@ -4,73 +4,28 @@
 
 namespace Chemaclass\JiraStatusNotifier\Channel;
 
-use function array_merge;
-use Chemaclass\JiraStatusNotifier\Jira\Board;
-use Chemaclass\JiraStatusNotifier\Jira\JiraHttpClient;
-use Chemaclass\JiraStatusNotifier\Jira\JqlUrlFactory;
 use Chemaclass\JiraStatusNotifier\Jira\ReadModel\JiraTicket;
-use function in_array;
 
 final class TicketsByAssignee
 {
-    private JiraHttpClient $jiraClient;
+    /** @psalm-var array<string, list<JiraTicket>> */
+    private array $list = [];
 
-    private JqlUrlFactory $jqlUrlFactory;
-
-    /** @var array */
-    private $jiraUsersToIgnore;
-
-    public function __construct(
-        JiraHttpClient $jiraClient,
-        JqlUrlFactory $jqlUrlFactory,
-        array $jiraUsersToIgnore
-    ) {
-        $this->jiraClient = $jiraClient;
-        $this->jqlUrlFactory = $jqlUrlFactory;
-        $this->jiraUsersToIgnore = $jiraUsersToIgnore;
-    }
-
-    /** @psalm-return array<string, array<array-key, JiraTicket>> */
-    public function fetchFromBoard(Board $board): array
+    public function add(JiraTicket $ticket): self
     {
-        $ticketsByAssignee = [];
-
-        foreach ($board->maxDaysInStatus() as $statusName => $maxDays) {
-            $grouped = $this->ticketsByAssigneeInStatus($statusName);
+        $assignee = $ticket->assignee();
 
-            foreach ($grouped as $assigneeKey => $tickets) {
-                if (!isset($ticketsByAssignee[$assigneeKey])) {
-                    $ticketsByAssignee[$assigneeKey] = $tickets;
-                } else {
-                    $ticketsByAssignee[$assigneeKey] = array_merge($ticketsByAssignee[$assigneeKey], $tickets);
-                }
-            }
+        if (!isset($this->list[$assignee->key()])) {
+            $this->list[$assignee->key()] = [];
         }
 
-        return $ticketsByAssignee;
+        $this->list[$assignee->key()][] = $ticket;
+
+        return $this;
     }
 
-    /** @psalm-return array<string, array<string, JiraTicket>> */
-    private function ticketsByAssigneeInStatus(string $statusName): array
+    public function list(): array
```

Comment:
maybe add the return type: `/** @psalm-return array<string, list<JiraTicket>> */`